pvlib/bifacial/infinite_sheds.py

```diff
@@ -139,19 +138,24 @@ def _vf_row_sky_integ(f_x, surface_tilt, gcr, npoints=100):
     """
     # handle Series inputs
     surface_tilt = np.array(surface_tilt)
-    cst = cosd(surface_tilt)
     # shaded portion
     x = np.linspace(0, f_x, num=npoints)
     psi_t_shaded = masking_angle(surface_tilt, gcr, x)
-    y = 0.5 * (cosd(psi_t_shaded) + cst)
+    y = 0.5 * (cosd(psi_t_shaded + surface_tilt) + 1)
     # integrate view factors from each point in the discretization. This is an
     # improvement over the algorithm described in [2]
-    vf_shade_sky_integ = np.trapz(y, x, axis=0)
+    vf_shade_sky_integ = np.where(
+        f_x<1e-6,
+        0.5 * (cosd(masking_angle(surface_tilt, gcr, 0.) + surface_tilt) + 1),
+        1 / f_x * np.trapz(y, x, axis=0))
     # unshaded portion
     x = np.linspace(f_x, 1., num=npoints)
     psi_t_unshaded = masking_angle(surface_tilt, gcr, x)
-    y = 0.5 * (cosd(psi_t_unshaded) + cst)
-    vf_noshade_sky_integ = np.trapz(y, x, axis=0)
+    y = 0.5 * (cosd(psi_t_unshaded + surface_tilt) + 1)
+    vf_noshade_sky_integ = np.where(
+        (1 - f_x)<1e-6,
+        0.5 * (cosd(surface_tilt) + 1),
```

Comment:
ditto, I think this might be `y[-1]` same reason as before:
```suggestion
        y[-1],
```

also here and before, perhaps a comment might explain that this avoids unnecessary divide by zero warnings and simplifies the calculation for case where shade line is either close to bottom or top of the panel